staging/src/k8s.io/apimachinery/pkg/watch/mux.go

```diff
@@ -218,6 +218,13 @@ func (m *Broadcaster) Action(action EventType, obj runtime.Object) {
 // if too many incoming actions are queued up.  Returns true if the action was sent,
 // false if dropped.
 func (m *Broadcaster) ActionOrDrop(action EventType, obj runtime.Object) bool {
+	// Ensure that if the broadcaster is stopped we do not send events to it.
+	select {
+	case <-m.stopped:
+		return false
+	default:
```

Comment:
Hrm yep I guess so... Maybe we have Shutdown() send a "special" message to the channel indicating that it needs to stop and cleanup, let me see what I can do

Reply:
I think that you can solve it with a mutex, so if Shutdown takes the lock Action has to wait or viceversa

Reply:
@deads2k you have more insights than anybody here per this discussion
https://github.com/kubernetes/kubernetes/pull/91602#discussion_r440857927

what do you think will be the best course of action , let it panic as today or  try to handle the shutdown better?

Reply:
I think there's a reasonable claim that the method signatures on Action, Watch, and WatchWithPrefix are incorrect and should all return errors if the broadcaster is shutdown.  Doing that would allow us to remove the panic.